consensus/dbft/dbft.go

```diff
@@ -1379,13 +1379,19 @@ func (c *DBFT) getBlockWitness(pub *tpke.PublicKey, block *Block) ([]byte, error
 						"from", i,
 						"error", err)
 				}
+				if v == dbftutil.ExtraV1 {
+					shares[dkgIndex].Neg()
+				}
 			}
 		}
 		msg := dbftRLP(block.header)
 		c.lock.RLock()
 		ks := c.amevKeystore
 		c.lock.RUnlock()
 		sig, err := ks.AggregateAndVerifySig(msg, shares)
+		if v == dbftutil.ExtraV1 {
+			sig.Neg()
+		}
```

Comment:
I'd like to abstract dBFT from math as much as possible. What do you think about:
```suggestion
		sig, err := ks.AggregateAndVerifySig(msg, shares, v == dbftutil.ExtraV1)
```

and then just something like:
```
func (ks *KeyStore) AggregateAndVerifySig(msg []byte, inputs map[int]*tpke.SignatureShare, negateResult bool) (*tpke.Signature, error) {
```

Reply:
And the same approach may be applied to `SignShare`.

Reply:
Good and possible.